cmdLine/src/main/scala/topgun/cmdline/FileParser.scala

```diff
@@ -121,16 +163,21 @@ class FileParser(file: File, cmdLine: JfrParseCommandLine, totals: Totals, confi
     }
   }
 
-  def cpu(event: RecordedEvent): Unit = {
+  def cpu(event: RecordedEvent, isNative:Boolean): Unit = {
+    if(isNative) {
+      totals.totalEvents.decrementAndGet()
+      totals.totalEventsNative.incrementAndGet()
+    }
     val thread = event.getThread
     if ((thread ne null) && !includeThread(thread.getJavaName)) {
-      totals.ignoredThreadCpuEvents.incrementAndGet
+      if(isNative) totals.ignoredThreadCpuEventsNative.incrementAndGet else totals.ignoredThreadCpuEvents.incrementAndGet
+
     } else {
       val distinctFrames: List[CallSite] = readFrames(event)
       if (!includeStack(distinctFrames)) {
-        totals.ignoredStackCpuEvents.incrementAndGet
+        if(isNative) totals.ignoredStackCpuEventsNative.incrementAndGet else totals.ignoredStackCpuEvents.incrementAndGet
```

Comment:
Do we know how much CPU time the tick relates to?
s it the same sample rate?

probably 
```
val counter = if (includeStack) 
  if (isNative)totals.consumedCpuEventsNative else totals.consumedCpuEvents
else   if (isNative)totals.ignoredStackCpuEventsNative else totals.ignoredStackCpuEvents
counter.incrementAndGet //or maybe counter.addAndGet(tickDuration)
```
 you can work out the tick duration in another PR :-)

Reply:
I need more explanation on this.

Reply:
work out the counter in one statement
then perform the operation